salt/metalk8s/kubeadm/init/preflight/mandatory.sls

```diff
@@ -0,0 +1,66 @@
+{%- from "metalk8s/map.jinja" import kubeadm_preflight with context %}
+{%- from "metalk8s/map.jinja" import defaults with context %}
+
+install mandatory packages:
+  pkg.installed:
+    - pkgs: {{ kubeadm_preflight.mandatory.packages }}
+
+{%- for service in kubeadm_preflight.mandatory.services.running %}
+enable {{ service  }} service:
+  service.running:
+    - name: {{ service  }}
+    - enable: true
+{%- endfor %}
+
+# required to set sysctl net.bridge.bridge-nf-call-iptables
+add the module br_netfilter to kernel:
+  kmod.present:
+    - name: br_netfilter
+    - persist: True
+
+{%- for item, value in kubeadm_preflight.mandatory.sysctl_values.items() %}
+set sysctl {{ item }} value to {{ value }}:
+  sysctl.present:
+    - name: {{ item }}
+    - value: {{ value }}
+    - config: /etc/sysctl.d/60-metalk8s.conf
+    {%- if item in ("net.bridge.bridge-nf-call-ip6tables", "net.bridge.bridge-nf-call-iptables") %}
+    - require:
+      - kmod: add the module br_netfilter to kernel
+    {%- endif %}
+{%- endfor %}
+
+{%- for swap_device in salt.mount.swaps().keys() %}
+disable swap device {{ swap_device }}:
+  module.run:
+    - name: mount.swapoff
+    - m_name: {{ swap_device }}
+{%- endfor %}
+
+prevent swap mount from fstab:
+  file.comment:
+    - name: /etc/fstab
+    - regex: .+\s+.+\s+swap\s+.+
+    - onlyif:
+      # Add a condition to avoid the state failure
+      # in case we don t have swap in /etc/fstab
+      - test -n "$(awk '$2=="swap" {print}' /etc/fstab)"
+
+check that the crictl socket answer:
+  cmd.run:
+    - name: crictl --runtime-endpoint unix:///run/containerd/containerd.sock info
```

Comment:
If we need `crictl` then we should add `cri-tools` as prerequisite

Reply:
Agree. Also, if we install `cri-tool`, we should create a configuration file for it such that one doesn't need to always specify `--runtime-endpoint` when invoking `crictl`.